File: android2_lesson06_2/src/java/java/com/example/ama/android2_lesson06_2/ui/list/adapter/DeviceListAdapter.java
package com.example.ama.android2_lesson06_2.ui.list.adapter;

import android.bluetooth.BluetoothDevice;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.ArrayList;

public class DeviceListAdapter extends ArrayAdapter<BluetoothDevice> {
    private ArrayList<BluetoothDevice> devices = new ArrayList<>();

    public DeviceListAdapter(@NonNull Context context) {
        super(context, 0);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext())
                    .inflate(android.R.layout.simple_list_item_1, parent, false);
        }
        ((TextView) convertView.findViewById(android.R.id.text1)).setText(devices.get(position).getName());
        return convertView;
    }

    public void addDevice(BluetoothDevice device) {
        if (!devices.contains(device)) {
            devices.add(device);
            super.clear();
            addAll(devices);
            notifyDataSetChanged();
        }
    }

    public void clear() {
        devices.clear();
        super.clear();
    }
}

File: android2_lesson06_2/src/kotlin/java/com/example/ama/android2_lesson06_2/ui/list/adapter/DeviceListAdapter.kt
package com.example.ama.android2_lesson06_2.ui.list.adapter

import android.bluetooth.BluetoothDevice
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class DeviceListAdapter(context: Context)
    : ArrayAdapter<BluetoothDevice>(context, android.R.layout.simple_list_item_1) {

    private var devices: ArrayList<BluetoothDevice> = ArrayList()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null) {
            view = LayoutInflater.from(context)
                    .inflate(android.R.layout.simple_list_item_1, parent, false)
        }
        view?.findViewById<TextView>(android.R.id.text1)?.text = devices[position].name
        return view!!
    }

    fun addDevice(device: BluetoothDevice) {
        if (!devices.contains(device)) {
            devices.add(device)
            super.clear()
            addAll(devices)
            notifyDataSetChanged()
        }
    }

    override fun clear() {
        devices.clear()
        super.clear()
    }
}